### PR TITLE
[G2M] Toast - timeout feature for connection hub

### PR DESCRIPTION
**Pull request by kc-leung**
- Added timeout feature for Toast component
- Added fade-in & fade-out animation

### CODE REVIEW

**Review by sallkall — APPROVED** [review]
If you have some extra time a simple css fade out animation would be even better @kc-leung

File: stories/toast.stories.js [hunk 1]
```diff
@@ -297,6 +293,8 @@ export const Usage = () => {
           button={buttonP}
           icon={<CheckBadge size='lg'/>} 
           onClose={() => setPopUp(false)}
+          timeOut={5000}
+          onTimeOut={() => setPopUp(false)}
```
Comment (hyx131) [inline stories/toast.stories.js:297]: `setPopUp => false` here is already called within the underlying `timeout` function so wouldn't need it here right, see above comment ^^

File: src/components/toast.js [hunk 2]
```diff
@@ -76,22 +78,42 @@ const Toast = forwardRef(({
     endIcon: `cursor-pointer stroke-current fill-current ${colorType[type].closeIcon}`,
   })
 
+  const [popUp, setPopUp] = useState(true)
+
+  useEffect(() => {
+    if (timeOut > 0) {
+      popUp && setTimeout(() => {
+        onTimeOut()
```
Comment (hyx131) [inline src/components/toast.js:86]: maybe do `if (onTimeOut) onTimeOut()` so users won't get any errors if they provided `timeOut` but doesn't need any extra actions so didn't need `onTimeOut`

**Review by hyx131 — APPROVED** [review]

File: src/components/toast.js [hunk 3]
```diff
@@ -10,13 +10,17 @@ import { concatTargetColor } from '../utils/concat-color'
 const Toast = forwardRef(({
   classes,
   variant,
+  display,
+  setDisplay,
```
Comment (hyx131) [inline src/components/toast.js:14]: would suggest not to include `setDisplay` as a prop, users can handle the `display -> true/false` externally, and via `onClose`

File: src/components/toast.js [hunk 4]
```diff
@@ -76,35 +80,79 @@ const Toast = forwardRef(({
...
+  let timer = ''
+  
+  useEffect(() => {
+    if (timeOut > 0) {
+      const toastEl = toastRef.current
+      let fade = ''
+      
+      display && (timer = setTimeout(() => {
+        if (onTimeOut) onTimeOut()
+        setDisplay(false)
```
Comment (hyx131) [inline src/components/toast.js:93]: so would also call `onClose` here instead of `setDisplay`

File: src/components/toast.js [hunk 5]
```diff
@@ -76,35 +80,79 @@ const Toast = forwardRef(({
...
+        }, 500)
+
+        if (fade) clearTimeout(fade)
+      }
+    }
+  }, [display])
+
+  const handleOnClose = () => {
+    onClose()
+    setDisplay(false)
```
Comment (hyx131) [inline src/components/toast.js:114]: can be handled by `onClose`

File: stories/toast.stories.js [hunk 6]
```diff
@@ -289,16 +287,16 @@ export const Usage = () => {
...
-          onClose={() => setPopUp(false)}
-        />
-      }
+      <Toast
+        type='semantic-light'
+        color='success'
+        title='Pop-up Success' 
+        button={buttonP}
+        icon={<CheckBadge size='lg'/>} 
+        display={popUp}
```
Comment (hyx131) [inline stories/toast.stories.js:296]: maybe can also rename `display` to `open` so it sounds more coherent with the close handler `onClose` lol

**Pull request merged**